dev/test/bulk-writer.ts

```diff
@@ -477,6 +482,146 @@ describe('BulkWriter', () => {
     });
   });
 
+  it('can retry failed create operations with retry()', async () => {
+    const bulkWriter = await instantiateInstance([
+      {
+        request: createRequest([createOp('doc', 'bar')]),
+        response: failedResponse(Status.FAILED_PRECONDITION),
+      },
+      {
+        request: createRequest([createOp('doc', 'bar')]),
+        response: successResponse(1),
+      },
+    ]);
+    let op: BulkWriterOperation | undefined = undefined;
+    bulkWriter.onError(async (error, operation) => {
+      op = operation;
+    });
+    bulkWriter.create(firestore.doc('collectionId/doc'), {foo: 'bar'});
+    await bulkWriter.flush();
+    expect(op).to.not.be.undefined;
+    let writeResult: WriteResult;
+    bulkWriter.retry(op!).then(res => {
+      writeResult = res;
+    });
+    return bulkWriter.close().then(() => {
+      expect(writeResult.writeTime.isEqual(new Timestamp(1, 0))).to.be.true;
+    });
+  });
+
+  it('can retry failed set operations with retry()', async () => {
```

Comment:
Do you think these duplicate tests add value? If so, you should try to extract helpers to reduce code duplication.